grpcwebclient/conformance-runner.ts

```diff
@@ -0,0 +1,205 @@
+// limitations under the License.
+
+import { browser } from "@wdio/globals";
+import * as esbuild from "esbuild";
+import { execFile } from "node:child_process";
+import type { Readable } from "node:stream";
+import * as net from "node:net";
+import { mkdtempSync } from "node:fs";
+import { tmpdir } from "node:os";
+import { join as joinPath } from "node:path";
+import {
+  ClientCompatRequest,
+  ClientCompatResponse,
+  ClientResponseResult,
+  ClientErrorResult,
+} from "./gen/proto/es/connectrpc/conformance/v1/client_compat_pb.js";
+import {
+  UnaryRequest,
+  ServerStreamRequest,
+  ClientStreamRequest,
+  BidiStreamRequest,
+  ConformancePayload_RequestInfo,
+  UnimplementedRequest,
+} from "./gen/proto/es/connectrpc/conformance/v1/service_pb.js";
+import { createRegistry } from "@bufbuild/protobuf";
+
+const typeRegistry = createRegistry(
+  UnaryRequest,
+  ServerStreamRequest,
+  ClientStreamRequest,
+  BidiStreamRequest,
+  ConformancePayload_RequestInfo,
+  UnimplementedRequest,
+  ClientCompatResponse,
+);
+
+describe("Connect Conformance", () => {
+  it("gRPC-web Client", async () => {
+    const buildResult = await esbuild.build({
+      entryPoints: ["./entry.ts"],
+      bundle: true,
+      write: false,
+    });
+    if (buildResult.outputFiles.length !== 1) {
+      throw new Error("Expected exactly one output file");
+    }
+    const invokeScript = buildResult.outputFiles[0].text;
+    const server = net.createServer((socket) => void run(socket, invokeScript));
+    let runnerCloseResolve: () => void,
+      runnerCloseReject: (err: unknown) => void;
+    const runnerClose = new Promise<void>((resolve, reject) => {
+      runnerCloseResolve = resolve;
+      runnerCloseReject = reject;
+    });
+    server.on("error", (err) => {
+      runnerCloseReject(err);
+    });
+
+    const socketName = joinPath(
+      mkdtempSync(joinPath(tmpdir(), "connectconformance")),
+      "socket",
+    );
+    server.listen(socketName, () => {
+      const runner = execFile(
+        "../.tmp/bin/connectconformance",
+        [
+          "--mode",
+          "client",
+          "--conf",
+          "../testdata/grpc-web-client-impl-config.yaml",
+          "--test-file",
+          "../internal/app/connectconformance/testsuites/grpc-web-client.yaml",
+          "-v",
+          "./bin/pipe",
+          socketName,
+        ],
+        {},
+      );
+      runner.stdout?.pipe(process.stdout);
+      runner.stderr?.pipe(process.stderr);
+      runner.on("error", (err) => {
+        runnerCloseReject(err);
+      });
+      runner.on("close", (code) => {
+        if (code !== 0) {
+          runnerCloseReject(new Error(`Runner exited with code ${code}`));
+        }
+        runnerCloseResolve();
+      });
+    });
+    await runnerClose.finally(() => server.close());
+
+    expect(true).toBe(true);
+  });
+});
+async function run(socket: net.Socket, invokeScript: string) {
```

Comment:
nit: mind adding a blank line above this, so the top-level elements are little more discernible at a glance?